tests/test_bids.py

```diff
@@ -1,116 +1,187 @@
+        cls.full_bidsmap_path = Path(cls.bidsmap_path.resolve())
+
+    def test_load_bidsmap(self):
+        # test loading with recommended arguments for load_bidsmap
+        full_arguments_map, return_path = bids.load_bidsmap(Path(self.full_bidsmap_path.name), self.full_bidsmap_path.parent)
+        self.assertIsInstance(full_arguments_map, dict)
+        self.assertTrue(full_arguments_map)
+
+        # test loading with no input folder0, should load default from heuristics folder
+        no_input_folder_map, _ = bids.load_bidsmap(self.bidsmap_path)
+        self.assertIsInstance(no_input_folder_map, dict)
+        self.assertTrue(no_input_folder_map)
+
+        # test loading with full path to only bidsmap file
+        full_path_to_bidsmap_map, _ = bids.load_bidsmap(self.full_bidsmap_path)
+        self.assertIsInstance(full_path_to_bidsmap_map, dict)
+        self.assertTrue(no_input_folder_map)
+
+    def test_find_run(self):
+        # load bidsmap
+
+        bidsmap, _ = bids.load_bidsmap(self.full_bidsmap_path)
+        # collect provenance from bidsmap for anat, pet, and func
+        anat_provenance = bidsmap['DICOM']['anat'][0]['provenance']
+        func_provenance = bidsmap['DICOM']['func'][0]['provenance']
+
+        # find run with partial provenance
+        not_found_run = bids.find_run(bidsmap=bidsmap, provenance='sub-001', dataformat='DICOM')
+        self.assertFalse(not_found_run)
+
+        # find run with full provenance
+        found_run = bids.find_run(bidsmap=bidsmap, provenance=anat_provenance)
+
+        # create a duplicate provenance but in a different datatype
+        bidsmap['PET'] = bidsmap['DICOM']
+        # mark the entry in the PET section to make sure we're getting the right one
+        tag = 123456789
+        bidsmap['PET']['anat'][0]['properties']['nrfiles'] = tag
+        # locate PET datatype run
+        pet_run = bids.find_run(bidsmap, provenance=anat_provenance, dataformat='PET')
+        self.assertEqual(
+            pet_run['properties']['nrfiles'], tag
+        )
+
+    def test_delete_run(self):
+        # create a copy of the bidsmap
+        with tempfile.TemporaryDirectory() as tempdir:
+            temp_bidsmap = Path(tempdir) / Path(self.full_bidsmap_path.name)
+            shutil.copy(self.full_bidsmap_path, temp_bidsmap)
+            bidsmap, _ = bids.load_bidsmap(temp_bidsmap)
+            anat_provenance = bidsmap['DICOM']['anat'][0]['provenance']
+            # now delete it from the bidsmap
+            bids.delete_run(bidsmap, anat_provenance)
+            self.assertEqual(len(bidsmap['DICOM']['anat']), 0)
+            # verify this gets deleted when rewritten
+            bids.save_bidsmap(_, bidsmap)
+            written_bidsmap, _ = bids.load_bidsmap(_)
+            deleted_run = bids.find_run(written_bidsmap, anat_provenance)
+            self.assertFalse(deleted_run)
+
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
As I mention in the initial PR comment, this was started shortly before pytest was included with bidscoin, this will get refactored to pytest from unittest as the former is just so much better.